### DISCUSSION TITLE
Noisy logs on credentials loading

### DISCUSSION THREAD

**Discussion by aalvrz** [boto/botocore#2662] [Q&A] [answered]
I am getting tons of logs from boto forwarded when credentials are being loaded. Example:

![image](https://user-images.githubusercontent.com/6859247/162809046-40d16a03-5dfd-4d35-915e-e4f774ac4d4a.png)

It seems to be because of this logging here:

https://github.com/boto/botocore/blob/5f0980812400f16cb10e71cf5abdeb27368fb0f7/botocore/credentials.py#L1114

Any ideas on how to suppress this? I am using boto3 usually like this:

```python
import boto3

session = boto3.session.Session(region_name=some_aws_region_config_var)
```

**Comment by tim-finnigan** [comment, marked as answer]
Hi @aalvrz here is the documentation on boto3 logging: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/boto3.html#boto3.set_stream_logger

You could customize which logging level to show, for example logging.WARNING. We don’t recommend logging in production as the logs could contain sensitive information.

**Comment by aalvrz** [comment]
@tim-finnigan Thanks for the response.

What is the correct way to disable all logging from boto3? Does `set_stream_logger` need to be called and set only once?

**Reply by tim-finnigan** [reply]
Logging should only be enabled if `boto3.set_stream_logger('')` exists in your code. But you could try something like this if you want to update the logging: https://github.com/boto/boto3/issues/521#issuecomment-653060090
